subprojects/core-api/src/main/java/org/gradle/api/file/FileTreeElement.java

```diff
@@ -73,8 +89,31 @@
      * @param target the target file.
      * @return true if this file was copied, false if it was up-to-date
      */
+    @Deprecated
     boolean copyTo(File target);
 
+    /**
+     * Copies this file to the given target {@link Path}. Does not copy the file if the target is already a copy of this file.
+     *
+     * @param target the target path.
+     * @return true if this file was copied, false if it was up-to-date
+     */
+    default boolean copyTo(Path target) {
+        try {
+            if (target.getParent() != null) {
+                Files.createDirectories(target.getParent());
+            }
+            if (!isDirectory()) {
+                Files.copy(getNioPath(), target, StandardCopyOption.COPY_ATTRIBUTES, StandardCopyOption.REPLACE_EXISTING, LinkOption.NOFOLLOW_LINKS);
```

Comment:
This means the actual copying does not agree with the up-to-date checks: For up-to-date checks we consider the _contents_ of the files (aka we follow the symlinks), while for the actual task action we wouldn't.

Reply:
@wolfs It is not clear to me whether you are stating what the current code in this PR does, or what your are expecting it do to (I blame English verb tenses).

Reply:
The code in this PR changes the logic so that the actual copying does not agree with up-to-date checks any more. Up-to-date checks follow symlinks and check the file contents, not target of the symlink.

I expect up-to-date checks to continue working as before. After this PR, if you change a symlink in the input to point to a different location with the same contents, then the task would be up-to-date. Though re-running it would create a different output.